apps/server/src/endpoints.ts

```diff
@@ -86,10 +88,20 @@ export const createGame =
       switch (body.modelType) {
         case ModelType.THREAT_DRAGON: {
           // TODO: validation
-          await gameServer.db.setModel(
-            matchID,
-            JSON.parse(body.model as string) as ThreatDragonModel,
-          );
+          logEvent('vishal printing body');
+          logEvent(`printing body: ${body.model}`);
+          var model = JSON.parse(body.model as string) as ThreatDragonModel;
+          await gameServer.db.setModel(matchID,model);
+          break;
+        }
+
+        case ModelType.THREAT_DRAGON_V2: {
+          // TODO: validation
+          logEvent('vishal printing body');
+          logEvent(`printing body: ${body.model}`);
+          var model2 = JSON.parse(body.model as string) as ThreatDragonModel2;
+          var model = mapModel2toOldModel(model2);
```

Comment:
Do I understand it correctly:  The Input is a V2 model, but internally it gets mapped to a V1 model? This would mean that the model that can be downloaded is also V1? For the users this might be every unintuitive...

Reply:
Ok. But if we use this as an interim solution, we should be transparent and rename the download button. Maybe add "(Threat Dragon V1)" tiny underneath the "Download Threats"?

Reply:
Yes @ChristophNiehoff  i have mapped v2 model to v1 as workaround. 
There is new diagram in V2 (box) which is handled. I am working on proper solution on v2.